libbpf-tools/bindsnoop.bpf.c

```diff
@@ -78,6 +88,12 @@ static int probe_exit(struct pt_regs *ctx, short ver)
 	if (filter_by_port && !port)
 		goto cleanup;
 
+	task = (struct task_struct*)bpf_get_current_task();
+	mntns_id = (u64) BPF_CORE_READ(task, nsproxy, mnt_ns, ns.inum);
+
+	if (filter_by_mnt_ns && !bpf_map_lookup_elem(&mount_ns_set, &mntns_id))
+		return 0;
```

Comment:
Should this be guarded by `filter_by_mnt_ns`? 

```suggestion
	if (filter_by_mnt_ns) {
		task = (struct task_struct*)bpf_get_current_task();
		mntns_id = (u64) BPF_CORE_READ(task, nsproxy, mnt_ns, ns.inum);

		if (!bpf_map_lookup_elem(&mount_ns_set, &mntns_id))
			return 0;
	}
```

Reply:
Well that's another way to write this. I think my way works fine and it relies on the compiler being smart enough to remove the non needed parts. Do you think we should rewrite it according to your suggestion?